frontend/public/components/utils/list-dropdown.jsx

```diff
@@ -177,10 +187,56 @@ ListDropdown.propTypes = {
 
 const NsDropdown_ = (props) => {
   const openshiftFlag = props.flags[FLAGS.OPENSHIFT];
+
   if (flagPending(openshiftFlag)) {
     return null;
   }
-  const kind = openshiftFlag ? 'Project' : 'Namespace';
+
+  const nsModel = modelFor('Namespace');
```

Comment:
I would suggest writing a simple hook or helper function that will give us the correct model. This will eliminate the need for conditionals later in the component. Within that hook, You can use the `useFlag` hook to retrieve the `CAN_CREATE_PROJECTS` flag that @spadgett suggested, and `useAccessReview` for namespaces. If user can't create either, then the result will be undefined and we can handle that in the logic below.

```tsx
 const model = useProjectOrNamespaceModel(/* args */);
```